addons/portal/controllers/binary.py

```diff
@@ -0,0 +1,63 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from werkzeug.exceptions import NotFound
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.mail.controllers.thread import ThreadController
+from odoo.addons.web.controllers.binary import Binary
+
+
+class BinaryController(Binary):
+    @http.route(
+        "/portal/thread/<string:thread_model>/<int:thread_id>/attachment/<int:attachment_id>",
+        methods=["GET"],
+        type="http",
+        auth="public",
+    )
+    def portal_thread_attachment(self, thread_model, thread_id, attachment_id, **kwargs):
+        if not ThreadController._get_thread_with_access(
+            thread_model, thread_id, mode=request.env[thread_model]._mail_post_access, **kwargs
+        ):
+            raise NotFound()
+        domain = [
+            ("id", "=", int(attachment_id)),
+            ("res_id", "=", int(thread_id)),
+            ("res_model", "=", thread_model),
+        ]
+        # sudo: ir.attachment - searching for an attachment on accessible thread is allowed
+        attachment_sudo = request.env["ir.attachment"].sudo().search(domain)
+        if not attachment_sudo:
+            raise NotFound()
+        return request.env["ir.binary"]._get_stream_from(attachment_sudo).get_response()
+
+    @http.route(
+        [
+            "/portal/thread/<string:thread_model>/<int:thread_id>/image/<int:attachment_id>",
+            "/portal/thread/<string:thread_model>/<int:thread_id>/image/<int:attachment_id>/<int:width>x<int:height>",
+        ],
+        methods=["GET"],
+        type="http",
+        auth="public",
+    )
+    def portal_fetch_image(
+        self, thread_model, thread_id, attachment_id, width=0, height=0, **kwargs
+    ):
+        if not ThreadController._get_thread_with_access(
```

Comment:
We no longer need this kind of route, this can be replaced by having `limited_field_access_token` also on attachments (same as avatar).

Reply:
And it doesn't need to be portal specific, we can also remove the binary controller of discuss channel by using the same trick in the base mail code.

Reply:
This is probably worth being a separate prep PR.

Reply:
odoo/odoo#204830